deploy/terraform/terraform-units/modules/sap_system/app_tier/vm-app.tf

```
@@ -74,7 +74,7 @@ resource "azurerm_linux_virtual_machine" "app" {
     [azurerm_network_interface.app[count.index].id]
   )
 
-  size                            = local.app_sizing.compute.vm_size
+  size                            = try(local.app_sizing.compute.vm_size, "Standard_D4s_v3")
```

If app_sizing can't be found, or if vm_size not defined in JSON, the deployment should fail rather than default vm_size to `Standard_D4s_v3`.

---

That was my initial thought as well - going back to that